udp_mux.go

```diff
@@ -365,13 +365,12 @@ func (m *UDPMuxDefault) connWorker() { //nolint:gocognit
 		var destinationConn *udpMuxedConn
 		m.addressMapMu.Lock()
 		if conns, ok := m.addressMap[addr.String()]; ok {
-			if localHost.IsUnspecified() {
+			destinationConn, ok = conns[ipAddr(localHost.String())]
+			if !ok {
 				for _, c := range conns {
```

Comment:
Dumb question? When there is no match, does it make sense to all the conns rather than just the first one?

Reply:
Dispatch to all the conns will cause repeated data packet received. Actually, I'm thinking of (and should do) re-design of udp mux, this fix PR will have same behavior as the old version(before my fix), both them can work but not ideal.

Reply:
Sounds good. Yeah, getting back to previous behaviour first is good.